### PR TITLE
CCT Support

### PR DESCRIPTION
**Pull request by jdsn**
This a proposal for the cct support like we discussed during our meeting on 2015-07-02.
extended by rick

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -3256,6 +3261,38 @@ function onadmin_qa_test()
...
+    addcloudrubygemrepo
+    # - install cct dependencies
+    ensure_packages_installed git-core gcc make ruby2.1-devel
+
+    # - default checkout location for mkcloud based setups:
+    mkdir -p /root/github.com/SUSE-Cloud
+    pushd /root/github.com/SUSE-Cloud/
+    git clone https://github.com/SUSE-Cloud/cct.git
+    cd cct
+    bundle install --path=.
```
Comment (vmoravec) [inline scripts/qa_crowbarsetup.sh]: The `--path` is not needed, `.bundle/config` contains `BUNDLE_PATH: vendor/bundle` and is loaded automatically by `bundler`

File: scripts/mkcloud [hunk 2]
```diff
@@ -121,6 +121,7 @@ scp="scp $sshopts"
 : ${localreposdir_target:="/repositories"}
 [ -z "$localreposdir_src" ] && localreposdir_target=""
 : ${install_chef_suse_override:=./install-chef-suse.sh}
+: ${cct_tests:=features}
```
Comment (vmoravec) [inline scripts/mkcloud]: I'm not sure about the `cct_tests` default name, it does not express what it does, I would rather change that to `base` and adapt the testsuite to provide `rake features:base`. Then it should be easy to combine multiple test categories like `base+storage+ha` and delegate that to respective `rake` tasks.

Comment (rsalevsky) [inline scripts/mkcloud]: Sure, can you adapt the testsuite?

Comment (vmoravec) [inline scripts/mkcloud]: Here it is: https://github.com/SUSE-Cloud/cct/pull/11

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -3256,6 +3261,38 @@ function onadmin_qa_test()
...
+        IFS='+'
+        for test in $cct_tests; do
+            bundle exec rake $test
+            if [[ $? -gt 0 ]]; then
+                ret=$?
+                popd
+                return $ret
+            fi
+        done
+        IFS=' '
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: :-1: the default IFS value is a different one, but you can simply avoid trouble by adding a "local IFS" line above the first assignment

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -3256,6 +3261,37 @@ function onadmin_qa_test()
...
+    cd cct
+    bundle install
+
+    if [[ -n $cct_tests ]]; then
+        local IFS
+        IFS='+'
+        for test in $cct_tests; do
+            bundle exec rake $test
+            if [[ $? -gt 0 ]]; then
+                ret=$?
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: :-1: in this place, $? is already the return code from the test above. just use if bundle exec ... ; then

**Pull request merged**